### PR TITLE
builtin: add random-bytes function

### PR DESCRIPTION
**Pull request by bom-d-van**

### CODE REVIEW

**Comment by coocood** [comment]
@bom-d-van 
The test is failing because that test case is unstable. I've submitted a PR to fix it.
You don't need to worry about it for now.

File: plan/typeinferer_test.go [hunk 1]
```diff
@@ -249,6 +249,7 @@ func (ts *testTypeInferrerSuite) TestInferType(c *C) {
 		{`sha1(123)`, mysql.TypeVarString, "utf8"},
 		{`sha(123)`, mysql.TypeVarString, "utf8"},
 		{`sha2(123, 256)`, mysql.TypeVarString, charset.CharsetUTF8},
+		{`random_bytes(32)`, mysql.TypeInt24, charset.CharsetBin},
```
Comment (coocood) [inline plan/typeinferer_test.go]: Should be `mysql.TypeVarString`, `charset.CharsetBin`.

Comment (bom-d-van) [inline plan/typeinferer_test.go]: Fixed! :)

**Comment by coocood** [comment]
@bom-d-van 
You just change the test case, but the typeinfer didn't change, so the test case failed.
```
FAIL: typeinferer_test.go:38: testTypeInferrerSuite.TestInferType
typeinferer_test.go:287:
    c.Assert(tp, Equals, ca.tp, Commentf("Tp for %s", ca.expr))
... obtained uint8 = 0x9
... expected uint8 = 0xfd
... Tp for random_bytes(32)
```

**Comment by bom-d-van** [comment]
@coocood Sorry for that, it should be alright now. Thanks!

**Comment by coocood** [comment]
LGTM

File: expression/builtin_encryption_test.go [hunk 2]
```diff
@@ -201,3 +201,32 @@ func (s *testEvaluatorSuite) TestMD5(c *C) {
...
+	f, err := fc.getFunction(datumsToConstants([]types.Datum{types.NewDatum(32)}), s.ctx)
+	c.Assert(err, IsNil)
+	out, err := f.eval(nil)
+	c.Assert(err, IsNil)
+	c.Assert(len(out.GetBytes()), Equals, 32)
+
+	f, err = fc.getFunction(datumsToConstants([]types.Datum{types.NewDatum(1025)}), s.ctx)
+	c.Assert(err, IsNil)
+	_, err = f.eval(nil)
+	c.Assert(err, Not(IsNil))
```
Comment (shenli) [inline expression/builtin_encryption_test.go]: You can use NotNil instead of Not(IsNil).

File: expression/builtin_encryption.go [hunk 3]
```diff
@@ -533,7 +534,26 @@ type builtinRandomBytesSig struct {
...
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return types.Datum{}, errors.Trace(err)
+	}
+	arg := args[0]
+	if arg.IsNull() {
+		return d, nil
+	}
+	size := arg.GetInt64()
+	if size < 1 || size > 1024 {
```
Comment (shenli) [inline expression/builtin_encryption.go]: MySQL returns the following error code and error message 
"mysql> select RANDOM_BYTES(2048);
ERROR 1690 (22003): length value is out of range in 'random_bytes'"

Please use standard error code or add a TODO comment here (Then we will fix it latter.).

Comment (bom-d-van) [inline expression/builtin_encryption.go]: I have changed it as you suggested. Thanks. :)

**Comment by tiancaiamao** [comment]
LGTM
Thanks for contribute! @bom-d-van

**Pull request merged**